### PR TITLE
refactor: remove sip10 form optional params

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/4616343550).

Remove optional params https://github.com/hirosystems/wallet/pull/3514#pullrequestreview-1371430437

### CODE REVIEW

File: src/app/pages/send/send-crypto-asset-form/form/stacks-sip10/stacks-sip10-fungible-token-send-form.tsx [hunk 1]
```diff
@@ -12,28 +12,42 @@ import { SendMaxButton } from '../../components/send-max-button';
 import { StacksCommonSendForm } from '../stacks/stacks-common-send-form';
 import { useSip10SendForm } from './use-sip10-send-form';
 
-export function StacksSip10FungibleTokenSendForm() {
+export function StacksSip10FungibleTokenSendFormSentinel() {
```
Comment (alter-eggo) [inline src/app/pages/send/send-crypto-asset-form/form/stacks-sip10/stacks-sip10-fungible-token-send-form.tsx]: not sure with naming here

File: src/app/query/stacks/balance/stacks-ft-balances.hooks.ts [hunk 2]
```diff
@@ -101,13 +103,20 @@ function useStacksFungibleTokenAssetBalancesUnanchoredWithMetadata(
...
-    return assetBalances.find(assetBalance => assetBalance.asset.contractId.includes(contractId));
-  }, [assetBalances, contractId]);
+    const balance = assetBalances.find(assetBalance =>
+      assetBalance.asset.contractId.includes(contractId)
+    );
+    if (!balance) {
+      toast.error('Unable to find balance by contract id');
+      navigate('..');
+    }
+    return balance as StacksFungibleTokenAssetBalance;
```
Comment (kyranjamie) [inline src/app/query/stacks/balance/stacks-ft-balances.hooks.ts:118]: Why is this type cast needed? If we `return` in the !balance doesn't it yield the same type?

File: src/app/pages/send/send-crypto-asset-form/form/stacks-sip10/stacks-sip10-fungible-token-send-form.tsx (hunk 1 above)
Comment (kyranjamie) [reply src/app/pages/send/send-crypto-asset-form/form/stacks-sip10/stacks-sip10-fungible-token-send-form.tsx]: Maybe the loader pattern/naming would work here, similar to home component?

Where loader takes care of the navigation when undefined, and passes to child component like 

https://github.com/hirosystems/wallet/blob/af69fd665d1ddb718f6ee949dc068a5d45b57a7f/src/app/pages/home/home.loader.tsx#L5-L13

pseudo ala

```ts
export function StacksSip10FungibleTokenSendForm() {
  return (
    <StacksSip10FtSendFormLoader>
      {({ contractId, symbol }) => // the rest of the component }
    </StacksSip10FtSendFormLoader>
  )
}
```

**Review by kyranjamie — APPROVED** [review]

**Review by fbwoolf — APPROVED** [review]

File: src/app/query/stacks/balance/stacks-ft-balances.hooks.ts (hunk 2 above)
Comment (alter-eggo) [reply src/app/query/stacks/balance/stacks-ft-balances.hooks.ts:118]: no( it returns `StacksFungibleTokenAssetBalance | undefined`, so it causes `undefined` errors in other hooks

File: src/app/pages/send/send-crypto-asset-form/form/stacks-sip10/stacks-sip10-fungible-token-send-form.tsx (hunk 1 above)
Comment (alter-eggo) [reply src/app/pages/send/send-crypto-asset-form/form/stacks-sip10/stacks-sip10-fungible-token-send-form.tsx]: done

**Pull request merged**